### DISCUSSION TITLE
Prisma support for distributed transactions

### DISCUSSION THREAD

**Discussion by philodi-dev** [prisma/prisma#19179] [Q&A] [answered]
Hello,

I am working on a distributed system architecture and each micro service has its own database.

I am using PostgreSQL with Prisma and Apollo Graphql.
Some features require interactive transactions; which I am able to support for a single microservice.

Unfortunately, some features and use cases require interactive transactions in two or three different microservices; which implies
as well accessing 2 or more database in a single micro service.

For example: 

- Create an account with userId 10 (trigger the user service).
- Create a default role and permissions for userId 10 (trigger the IAM service)
- Create a wallet for userId 10 (trigger wallet service)

For the example above; I need them to succeed or fail together but I can't run interactive transactions since they are 3 different databases.

I managed to accomplish this using a message broker like Kafka but it resulted in performance issues, deadlocks, involvement in network traffic, lots of code, ...

I would like to know if there is a way to handle this using Prisma, for example by supporting the connection of multiple data sources.

In short, is it possible to connect to 2 or more different databases using prisms via a single miroservice?
I can not find anything related on the official documentation.

Regards,
Phil

**Comment by nurul3101** [comment, marked as answer]
Hello @philodi-dev 👋 

We do have a feature request to support connecting to Multiple Databases:

- https://github.com/prisma/prisma/issues/2443

You would need to create multiple PrismaClient instances to connect to different databases, as mentioned in this [comment](https://github.com/prisma/prisma/issues/2443#issuecomment-630679118).

Though Prisma doesn't support distributed transactions natively yet.

**Reply by nurul3101** [reply]
@philodi-dev If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
